onadata/libs/serializers/data_serializer.py

```diff
@@ -303,3 +303,48 @@ def create(self, validated_data):
                                      validated_data['id_string'])
 
         return instance
+
+
+class FLOIPSubmissionSerializer(SubmissionSuccessMixin,
+                                serializers.Serializer):
+    """
+    FLOIP SubmmissionSerializer - Handles a row of FLOIP specification format.
+    """
+    def validate(self, attrs):
+        data = self.context['request'].data
+        error_msg = None
+
+        if not isinstance(data, list):
+            error_msg = u'Invalid format. Expecting a list.'
+        else:
+            for row_i in xrange(len(data)):
+                if len(data[row_i]) != 6:
+                    error_msg = u"All rows must have 6 values, "\
+                                u"row %s does not." % row_i
+                    break
+
+        if error_msg:
+            raise serializers.ValidationError(_(error_msg))
+
+        return super(FLOIPSubmissionSerializer, self).validate(attrs)
+
+    def to_internal_value(self, data):
+        """
+        Overrides validating rows in list data.
+        """
+        data = {data[1]: data}
+
+        return data
+
+    def create(self, validated_data):
+        """
+        Returns object instances based on the validated data.
+        """
+        request, username = get_request_and_username(self.context)
+        xform_pk = self.context['view'].kwargs['xform_pk']
+        xform_id = get_object_or_404(XForm, pk=xform_pk).id_string
+        xform_headers = XForm.objects.get(id_string=xform_id).get_keys()
```

Comment:
Why to queries here when we can do with only one?

Do note that `id_string` is not a unique field, it is possible to have the same id_string in different user accounts.

```python
xform = get_object_or_404(XForm, pk=xform_pk)
xform_headers = xform.get_keys()
```